src/components/i18n/DopplerIntlProvider.test.tsx

```diff
@@ -1,33 +1,77 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
 import { DopplerIntlProvider } from "./DopplerIntlProvider";
-import { render } from "@testing-library/react";
-import "@testing-library/jest-dom/extend-expect";
 import { FormattedMessage } from "react-intl";
+import { useAppSessionState } from "../AppSessionStateContext";
+import { AppSessionState } from "../../abstractions/app-session";
 
-describe("DopplerIntlProvider", () => {
-  it("should render Campaign in English", () => {
-    const { getByText } = render(
-      <DopplerIntlProvider locale="en">
-        <FormattedMessage id="campaign" />
-      </DopplerIntlProvider>
-    );
-    getByText("Campaign");
-  });
+jest.mock("../AppSessionStateContext");
+jest.mock("./en", () => ({
+  messages_en: {
+    lang: "en",
+  },
+}));
+jest.mock("./es", () => ({
+  messages_es: {
+    lang: "es",
+  },
+}));
 
-  it("should render Campaign in Spanish", () => {
-    const { getByText } = render(
-      <DopplerIntlProvider locale="es">
-        <FormattedMessage id="campaign" />
-      </DopplerIntlProvider>
-    );
-    getByText("Campaña");
-  });
+const UNKNOWN_SESSION: AppSessionState = {
+  status: "unknown",
+};
+const AUTHENTICATED: AppSessionState = {
+  lang: "en",
+  status: "authenticated",
+  jwtToken: "jwt",
+  unlayerUser: {
+    id: "000",
+    signature: "000",
+  },
+  dopplerAccountName: "doppler_mock@mail.com",
+};
+const AUTHENTICATED_WITH_LANG: any | AppSessionState = {
+  status: "authenticated",
+  jwtToken: "jwt",
+  unlayerUser: {
+    id: "000",
+    signature: "000",
+  },
+  dopplerAccountName: "doppler_mock@mail.com",
+};
+const LANG_DEFAULT = "es";
 
-  it("should render Campaign in Spanish when language is unexpected", () => {
-    const { getByText } = render(
-      <DopplerIntlProvider locale="fr">
-        <FormattedMessage id="campaign" />
-      </DopplerIntlProvider>
-    );
-    getByText("Campaña");
-  });
+beforeEach(() => {
+  (useAppSessionState as jest.Mock).mockImplementation(() => UNKNOWN_SESSION);
+});
+
+it("should translated using the default language", () => {
```

Comment:
who is `it`? I think that `describe is missing`

Reply:
Now the using of describe isn't significance in the code because we use the testing file name for describe. e.g https://github.com/FromDoppler/doppler-editors-webapp/blob/a1e55a0685b88bf931e0c2a6514511a238a4b6a7/src/components/Editor.test.tsx#L50

We can talk about it in a meeting.